<launch>
  <arg name="world_name" default="worlds/empty.world"/>
  <arg name="limited" default="false"/>
  <arg name="gui" default="true" doc="Starts gazebo gui" />
  
  <!-- startup simulated world -->
  <include file="$(find ur_digital_twin)/launch/setup_world.launch">
    <arg name="gui" value="$(arg gui)"/>
    <arg name="physics" value="ode"/>
    <arg name="world_name" value="$(arg world_name)"/>
  </include>

  <!-- send robot urdf to param server -->
  <include file="$(find ur_description)/launch/ur5_upload.launch">
    <arg name="limited" value="$(arg limited)"/>
  </include>

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model robot -z 1.3 
      -J shoulder_pan_joint 0.0
      -J shoulder_lift_joint -1.5707
      -J elbow_joint 1.5707
      -J wrist_1_joint -1.5707
      -J wrist_2_joint -1.5707
      -J wrist_3_joint 0.0" respawn="false" output="screen" />

  <include file="$(find ur_gazebo)/launch/controller_utils.launch"/>

  <!-- start this controller -->
  <rosparam file="$(find ur_gazebo)/controller/arm_controller_ur5.yaml" command="load"/>
  <node name="arm_controller_spawner" pkg="controller_manager" type="controller_manager" args="spawn arm_controller" respawn="false" output="screen"/>

  <!-- load other controllers -->
  <node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false" output="screen" args="load joint_group_position_controller" />
</launch>